Client/src/Components/ProtectedRoute/index.jsx

```diff
@@ -3,21 +3,20 @@ import { useSelector } from "react-redux";
 import PropTypes from "prop-types";
 
 /**
- * ProtectedRoute is a higher-order component that wraps the `Route` component from `react-router-dom`
- * to create a protected route. It uses Redux to check if the user is authenticated. If the user is
- * authenticated, it renders the component passed to it; otherwise, it redirects the user to the login page.
+ * ProtectedRoute is a wrapper component that ensures only authenticated users
+ * can access the wrapped content. It checks authentication status (e.g., from Redux or Context).
+ * If the user is authenticated, it renders the children; otherwise, it redirects to the login page.
  *
  * @param {Object} props - The props passed to the ProtectedRoute component.
- * @param {React.ComponentType} props.Component - The component to render if the user is authenticated.
- * @param {Object} rest - The remaining props passed to the Route component.
- * @returns {React.ReactElement} A Route component that conditionally renders the passed Component or redirects to the login page.
+ * @param {React.ReactNode} props.children - The children to render if the user is authenticated.
+ * @returns {React.ReactElement} The children wrapped in a protected route or a redirect to the login page.
  */
 
-const ProtectedRoute = ({ Component, ...rest }) => {
+const ProtectedRoute = ({ children }) => {
 	const authState = useSelector((state) => state.auth);
 
 	return authState.authToken ? (
-		<Component {...rest} />
+		children
```

Comment:
It won't change much, but it leaves clearer which props belong to whom.
Our current approach would be something called render props, in which we pass a component as a prop, and we would have to pass that component's props to the parent component. 
Using children we can:
- Pass more than one component
- Pass each component's props to them, instead of passing to the parent, that needs to pass them down

I find it more simple to read also